app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -0,0 +1,9 @@
+module ManageIQ::Providers::Azure::CloudManager::Provision::StateMachine
+  def customize_destination
```

Comment:
ok, thanks. @bronaghs and me thought exactly this. So it might be worth either introduce a new step called `associate_floating_ip` or move this floating ip logic into the provider code. I'll do this in a follow up PR